packages/page-explorer/src/BlockInfo/Summary.tsx

```diff
@@ -10,41 +11,47 @@ import { CardSummary, SummaryBox } from '@polkadot/react-components';
 import { useApi } from '@polkadot/react-hooks';
 import { convertWeight } from '@polkadot/react-hooks/useWeight';
 import { FormatBalance } from '@polkadot/react-query';
-import { BN, BN_ONE, BN_THREE, BN_TWO, formatNumber } from '@polkadot/util';
+import { BN, BN_ONE, BN_THREE, BN_TWO, BN_ZERO, formatNumber } from '@polkadot/util';
 
 import { useTranslation } from '../translate.js';
 
 interface Props {
   events?: KeyedEvent[] | null;
   maxBlockWeight?: BN;
+  maxProofSize?: BN;
   signedBlock?: SignedBlock;
 }
 
-function extractEventDetails (events?: KeyedEvent[] | null): [BN?, BN?, BN?] {
+function extractEventDetails (events?: KeyedEvent[] | null): [BN?, BN?, BN?, BN?] {
   return events
-    ? events.reduce(([deposits, transfers, weight], { record: { event: { data, method, section } } }) => [
-      section === 'balances' && method === 'Deposit'
-        ? deposits.iadd(data[1] as Balance)
-        : deposits,
-      section === 'balances' && method === 'Transfer'
-        ? transfers.iadd(data[2] as Balance)
-        : transfers,
-      section === 'system' && ['ExtrinsicFailed', 'ExtrinsicSuccess'].includes(method)
-        ? weight.iadd(
-          convertWeight(
+    ? events.reduce(([deposits, transfers, weight], { record: { event: { data, method, section } } }) => {
+      return [
+        section === 'balances' && method === 'Deposit'
+          ? deposits.iadd(data[1] as Balance)
+          : deposits,
+        section === 'balances' && method === 'Transfer'
+          ? transfers.iadd(data[2] as Balance)
+          : transfers,
+        section === 'system' && ['ExtrinsicFailed', 'ExtrinsicSuccess'].includes(method)
+          ? weight.iadd(convertWeight(
             ((method === 'ExtrinsicSuccess' ? data[0] : data[1]) as DispatchInfo).weight
-          ).v1Weight
-        )
-        : weight
-    ], [new BN(0), new BN(0), new BN(0)])
+          ).v1Weight)
+          : weight,
+        section === 'system' && ['ExtrinsicFailed', 'ExtrinsicSuccess'].includes(method)
+          ? (convertWeight(
+            ((method === 'ExtrinsicSuccess' ? data[0] : data[1]) as DispatchInfo).weight
+          ).v2Weight as V2Weight).proofSize.toBn()
```

Comment:
@TarikGul could you please double-check this part? I'm not sure I might be missing some more weight here.

Reply:
Looks solid to me